submit/submit

```diff
@@ -0,0 +1,534 @@
+        error('REST API returned unexpected JSON data')
+
+    print('Submission received, id = s{}'.format(submission['id']))
+
+version_text = '''
+submit -- part of DOMjudge
+Written by the DOMjudge developers
+
+DOMjudge comes with ABSOLUTELY NO WARRANTY.  This is free software, and you
+are welcome to redistribute it under certain conditions.  See the GNU
+General Public Licence for details.
+'''
+
+loglevels = {
+    'DEBUG': logging.DEBUG,
+    'INFO': logging.INFO,
+    'WARNING': logging.WARNING,
+    'ERROR': logging.ERROR,
+    'CRITICAL': logging.CRITICAL,
+}
+
+# Note: we set add_help to false since we can only print the help text after parsing flags, since the help contains data needed from the API
+parser = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter, description='Submit a solution for a problem.', add_help=False)
+parser.add_argument('--version', action='version', version=version_text, help='output version information and exit')
+parser.add_argument('-h', '--help', help='display this help and exit', action='store_true')
+parser.add_argument('-c', '--contest', help='''submit for contest with ID or short name CONTEST.
+    Defaults to the value of the
+    environment variable 'SUBMITCONTEST'.
+    Mandatory when more than one contest is active.''')
+parser.add_argument('-p', '--problem', help='submit for problem with ID or label PROBLEM')
+parser.add_argument('-l', '--language', help='submit in language with ID LANGUAGE')
+parser.add_argument('-e', '--entry_point', help='set an explicit entry_point, e.g. the java main class')
+parser.add_argument('-v', '--verbose', help='increase verbosity', choices=loglevels.keys(), nargs='?', const='INFO', default='WARNING')
+parser.add_argument('-q', '--quiet', help='suppress warning/info messages', action='store_true')
+parser.add_argument('-y', '--assume-yes', help='suppress user input and assume yes', action='store_true')
+parser.add_argument('-u', '--url', help='''submit to server with base address URL
+    (should not be necessary for normal use)''')
+parser.add_argument('filename', nargs='*', help='filename(s) to submit')
+
+args = parser.parse_args()
+
+verbosity = args.verbose
+if args.quiet:
+    verbosity = 'ERROR'
+
+logging.basicConfig(format='[%(asctime)s] {}[{}]: %(message)s'.format(sys.argv[0], os.getpid()), datefmt='%b %d %H:%M:%S', level=loglevels[verbosity])
+
+contest_id = ''
+
+if 'SUBMITBASEURL' in os.environ:
+    baseurl = os.environ['SUBMITBASEURL']
+if 'SUBMITCONTEST' in os.environ:
+    contest_id = os.environ['SUBMITCONTEST']
+
+if args.contest is not None:
+    contest_id = args.contest
+problem_id = args.problem or ''
+language_id = args.language or ''
+entry_point = args.entry_point
+if args.url:
+    baseurl = args.url
+
+logging.info('set verbosity to {}'.format(verbosity))
+
+# Make sure that baseurl terminates with a '/' for later concatenation.
+if baseurl != '' and baseurl[-1:] != '/':
```

Comment:
Does it make sense to continue at all if baseurl is not set after this point?

Reply:
The main problem with this, and a lot of your comments below: the 'epilog' for the argument list is calculated using variables we only determine below. If we would stop here, you can't render the epilog so `--help` would show way less. Is that something that is acceptable? Note: this is how the C++ version did it, but I am totally fine with changing this to make the flow better.

Reply:
Ok, let's keep it like this and I'll stare hard at it when it's committed